lambdas/select-cri/src/main/java/uk/gov/di/ipv/core/selectcri/SelectCriHandler.java

```diff
@@ -217,6 +210,64 @@ private APIGatewayProxyResponseEvent getJourneyResponse(String criId) {
                 HttpStatus.SC_OK, new JourneyResponse(String.format(CRI_START_JOURNEY, criId)));
     }
 
+    private APIGatewayProxyResponseEvent getJourneyPyiNoMatchResponse() {
+        return ApiGatewayResponseGenerator.proxyJsonResponse(
+                HttpStatus.SC_OK, new JourneyResponse("/journey/pyi-no-match"));
+    }
+
+    private APIGatewayProxyResponseEvent getJourneyKbvFailResponse() {
+        return ApiGatewayResponseGenerator.proxyJsonResponse(
+                HttpStatus.SC_OK, new JourneyResponse("/journey/pyi-kbv-fail"));
+    }
+
+    private Optional<APIGatewayProxyResponseEvent> getCriResponse(
+            List<VisitedCredentialIssuerDetailsDto> visitedCredentialIssuers,
+            List<VcStatusDto> currentVcStatuses,
+            String criId,
+            String userId)
+            throws ParseException {
+        CredentialIssuerConfig criConfig = configurationService.getCredentialIssuer(criId);
+
+        LOGGER.info("CRI config audience value: {}", criConfig.getAudienceForClients());
+
+        Optional<VcStatusDto> vc = getVc(currentVcStatuses, criConfig.getAudienceForClients());
+        if (vc.isEmpty()) {
+            if (userHasNotVisited(visitedCredentialIssuers, criId)) {
+                return Optional.of(getJourneyResponse(criId));
+            }
+
+            if (criId.equals(dcmawCriId)) {
+                LOGGER.info("Routing user to web journey");
+                return Optional.of(
+                        getNextWebJourneyCri(visitedCredentialIssuers, currentVcStatuses, userId));
```

Comment:
What's the scenario where a user doesn't has a VC status (returned by getVc()), but _has_ visited the CRI?

Reply:
So its if we a user gets an Oauth error from a CRI. In that scenario they have visited the CRI but won't have a VC. 

Specifically this kind of thing will happen in the app. If the user decides to not continune on the app journey after we send them there, the app will send the user back with an `access_denied` oauth error. The KBV cri is also going to be doing some stuff with oauth errors as well